modules/apps/portal-language/portal-language-override-service/src/main/java/com/liferay/portal/language/override/service/impl/PLOEntryLocalServiceImpl.java

```diff
@@ -118,6 +127,33 @@ public int getPLOEntriesCount(long companyId) {
 		return ploEntryPersistence.countByCompanyId(companyId);
 	}
 
+	@Override
+	public void importPLOEntries(
+			long companyId, InputStream inputStream, String languageId,
+			long userId)
+		throws IOException, PortalException {
+
+		Properties properties = new Properties();
+
+		properties.load(
+			new InputStreamReader(inputStream, StandardCharsets.UTF_8));
+
+		if (properties.isEmpty()) {
+			throw new InvalidPropertiesFileException("Invalid properties file");
+		}
+
+		Enumeration<String> enumeration =
+			(Enumeration<String>)properties.propertyNames();
+
+		while (enumeration.hasMoreElements()) {
+			String key = enumeration.nextElement();
+
+			addOrUpdatePLOEntry(
+				companyId, userId, key, languageId,
+				properties.getProperty(key));
+		}
```

Comment:
I thought we were going to collect all errors _then_ fail instead of failing on the first one. Did I misunderstand?

Reply:
We shifted our focus to the rollback, so I don't remember to be honest, so I just kept the way it was. I'll collect all errors

Reply:
Done, we now have `ImportTranslationsException` which contains two internal classes `InvalidPropertiesFile` and `InvalidTranslations`